src/array/cuda/sddmm.cuh

```diff
@@ -86,7 +86,12 @@ namespace cuda {
     }                                                                    \
   } while (0)
 
+#ifdef __HIPCC__
+// HIP requires 64-bit mask
```

Comment:
Alternatively, if you know the __syncXXX function is going to use a full mask, you can just call the version of it that doesn't use a mask.  That's what pytorch has done.  See this file.

https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/cuda/DeviceUtils.cuh

Reply:
I know there was a plan to support these *_sync APIs for HIP even though they don't make much sense for our HW.  So perhaps they're supported now and your ifdefing is working and the preferred approach.